cmd/snap-confine/snap-confine.c

```diff
@@ -112,6 +112,10 @@ int main(int argc, char **argv)
 #endif				// ifdef HAVE_SECCOMP
 
 	if (geteuid() == 0) {
+		// ensure that "/" or "/snap" is mounted with the
```

Comment:
Hm, sc_initialize_ns_groups() feels a bit strange (conceptually). Also it requires that discard_ns links with mount-support which in turn requires libcap for discard-mount-support it seems. Nothing that cannot be fixed still feels a bit awkward. I wonder what is the worst that can happen if we don't lock - it seems the worst would be a double bind mount?